benchmark/Benchmark.h

```diff
@@ -40,58 +42,113 @@ class BenchmarkRecords {
       // The names of the columns and rows.
       std::vector<std::string> rowNames;
       std::vector<std::string> columnNames;
-      // The actual content of the fields. As in the seconds one function
-      // execution takes. Access is [row, column].
-      std::vector<std::vector<float>> entries;
+      // The time measurements in seconds. Access is [row, column].
+      std::vector<std::vector<std::optional<float>>> entries;
+
+      // Default constructor. Without it, some of the included headers generate
+      // errors.
+      RecordTable(): descriptor{""}, rowNames{{}}, columnNames{{}}, entries{{}} {}
+
+      RecordTable(const std::string& pDescriptor, 
+          const std::vector<std::string>& pRowNames,
+          const std::vector<std::string>& pColumnNames):
+          descriptor{pDescriptor}, rowNames{pRowNames},
+          columnNames{pColumnNames},
+          entries(rowNames.size(), std::vector<std::optional<float>>(rowNames.size())) {}
     };
 
   private:
 
     // A vector of all single functions measured.
-    std::vector<RecordEntry> _singleMeasurments;
+    std::vector<RecordEntry> singleMeasurements_;
 
     // A hash map of all the created RecordGroups. For faster access.
     // The key for a RecordGroup is it's descriptor.
-    ad_utility::HashMap<std::string, RecordGroup> _recordGroups;
+    ad_utility::HashMap<std::string, RecordGroup> recordGroups_;
+    // The order of creation for the entries of recordGroups_.
+    std::vector<std::string> recordGroupsOrder_;
   
     // A hash map of all the created RecordTables. For faster access.
     // The key for a RecordTable is it's descriptor.
-    ad_utility::HashMap<std::string, RecordTable> _recordTables;
-    
+    ad_utility::HashMap<std::string, RecordTable> recordTables_;
+     // The order of creation for the entries of recordTables_.
+    std::vector<std::string> recordTablesOrder_;
+
```

Comment:
This very much looks like it could be a templated class that keeps
the hashMap and the vector in sync, ensures that there are no duplicates, and owns the functionality to get the members in order.
Maybe call it `HashMapWithInsertionOrder` and implement it in a different file.

Reply:
Done, see [28da74a](https://github.com/ad-freiburg/qlever/pull/860/commits/28da74a07d067d6738925eb47645e132ef0ad46a).